warehouse/core/src/main/java/datawave/iterators/filter/ageoff/LatestVersionFilter.java

```diff
@@ -0,0 +1,396 @@
+     *            a non-null filterConfiguration
+     * @return whether to accept or reject this timestamp
+     */
+    @VisibleForTesting
+    boolean accept(long dataTimestamp, VersionFilterConfiguration filterConfiguration) {
+        // determine dataType's configured timestamp
+        long configuredTimestamp = filterConfiguration.timestampVersion;
+        
+        // apply logic for configured mode
+        switch (filterConfiguration.mode) {
+            case EQUAL:
+                return dataTimestamp == configuredTimestamp;
+            case GREATER_THAN_OR_EQUAL:
+                return dataTimestamp >= configuredTimestamp;
+            case UNDEFINED:
+            default:
+                log.warn("DataType {} was not configured for 'eq' or 'gte'");
+                // By default: accept all data at compaction scopes, no data at scan scope
+                return !this.isScanScope;
+        }
+    }
+    
+    /**
+     * @param source
+     *            simply passed to super.init
+     * @param options
+     *            examined to create filterConfiguration, isIndexTable
+     * @param iterEnv
+     *            examined to create filterConfiguration, isIndexTable, isScanScope
+     * @throws IOException
+     *             only expected from super.init
+     */
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment iterEnv) throws IOException {
+        super.init(source, options, iterEnv);
+        
+        // determine if this is operating at scan
+        this.isScanScope = (IteratorUtil.IteratorScope.scan == iterEnv.getIteratorScope());
+        
+        this.isIndexTable = extractIndexTableStatus(options, iterEnv);
+        
+        this.dataTypeConfigurations = gatherDataTypeConfigurations(options, iterEnv);
+    }
+    
+    /**
+     * Examines the Map of iterator options and IteratorEnvironment to create a mapping between data types and their filter configurations
+     * 
+     * @param options
+     *            iterator options
+     * @param iterEnv
+     *            iterator environment
+     * @return map containing dataType to VersionFilterConfiguration for its data
+     */
+    private Map<ByteSequence,VersionFilterConfiguration> gatherDataTypeConfigurations(Map<String,String> options, IteratorEnvironment iterEnv) {
+        Map<ByteSequence,VersionFilterConfiguration> configurationsMap = new HashMap<>();
+        
+        String dataTypeListOptionValue = options.get(DATATYPE_LIST_OPTION_NAME);
+        if (Strings.isNullOrEmpty(dataTypeListOptionValue)) {
+            log.trace(DATATYPE_LIST_OPTION_NAME + " is null or empty.  Configure data type list.");
+            return configurationsMap;
+        }
+        
+        String[] dataTypes = StringUtils.split(dataTypeListOptionValue, ",");
+        for (String dataTypeStr : dataTypes) {
+            if (null == dataTypeStr || dataTypeStr.trim().length() == 0) {
```

Comment:
Would Strings.isNullOrEmpty() make more sense here?

Reply:
Strings.isNullOrEmpty doesn't do the trim